warp10/src/main/sh/warp10.sh

```diff
@@ -17,506 +17,372 @@
-  secret=`su ${WARP10_USER} -c "${JAVACMD} -cp ${WARP10_JAR} -Dlog4j.configuration=file:${LOG4J_CONF} -Dfile.encoding=UTF-8 io.warp10.WarpConfig ${CONFIG_FILES} . 'token.secret' | grep -e '^@CONF@ ' | sed -e 's/^@CONF@ //' | grep 'token.secret' | sed -e 's/^.*=//'"`
-  if [[ "${secret}"  != "null" ]]; then
-    sed -i${SED_SUFFIX} -e "s|^{{secret}}|'"${secret}"'|" ${WARP10_HOME}/templates/warp10-tokengen.mc2
-  else
-    sed -i${SED_SUFFIX} -e "s|^{{secret}}||" ${WARP10_HOME}/templates/warp10-tokengen.mc2
+  # shellcheck disable=SC2086
+  if ! ${JAVACMD} -cp "${WARP10_JAR}" -Dfile.encoding=UTF-8 io.warp10.GenerateCryptoKey ${CONFIG_FILES}; then
+    die "ERROR: Failed to generated AES and hash keys"
   fi
-  rm ${WARP10_HOME}/templates/warp10-tokengen.mc2${SED_SUFFIX}
-
-  # Generate read/write tokens valid for a period of 100 years. We use 'io.warp10.bootstrap' as application name.
-  su ${WARP10_USER} -c "${JAVACMD} -cp ${WARP10_JAR} -Dlog4j.configuration=file:${LOG4J_CONF} -Dfile.encoding=UTF-8 io.warp10.TokenGen ${CONFIG_FILES} ${WARP10_HOME}/templates/warp10-tokengen.mc2" >  ${WARP10_HOME}/etc/initial.tokens
-  sed -i${SED_SUFFIX} 's/^.\{1\}//;$ s/.$//' ${WARP10_HOME}/etc/initial.tokens # Remove first and last character
-  rm "${WARP10_HOME}/etc/initial.tokens${SED_SUFFIX}"
 
-  echo "Warp 10 config has been generated here: ${WARP10_CONFIG_DIR}"
-
-  echo "You can now configure the initial and maximum amount of RAM allocated to Warp 10. Edit ${WARP10_HOME}/bin/warp10-env.sh, and look for WARP10_HEAP and WARP10_HEAP_MAX variables."
+  # Fix ownership
+  echo "Fix ownership"
+  chown -hRH "${WARP10_USER}:${WARP10_GROUP}" "${WARP10_HOME}"
+  chown "${WARP10_USER}:${WARP10_GROUP}" "${WARP10_HOME}"
+  if [ -n "${WARP10_DATA_DIR:-}" ]; then
+    chown -hRH "${WARP10_USER}:${WARP10_GROUP}" "${WARP10_DATA_DIR}"
+  fi
 
-  touch ${FIRSTINIT_FILE}
+  echo
+  echo "Warp 10 configuration has been generated here: ${WARP10_CONFIG_DIR}"
+  echo "You can now configure the initial and maximum amount of RAM allocated to Warp 10."
+  echo "Edit ${WARP10_HOME}/bin/warp10-env.sh and look for WARP10_HEAP and WARP10_HEAP_MAX variables."
+  echo
+}
 
+distConf() {
+  echo "Initialize Warp 10 distributed configuration"
+  init
+  postInit
 }
 
-start() {
+leveldbConf() {
+  echo "Initialize Warp 10 standalone configuration"
+  init
 
-  #
-  # Make sure the caller is WARP10_USER
-  #
-  isUser ${WARP10_USER}
+  sed -i -e 's|^standalone\.home.*|standalone.home = '"${WARP10_HOME_ESCAPED}"'|' "${WARP10_CONFIG_DIR}/00-warp.conf"
+  sed -i -e 's|^backend = .*|backend = leveldb|' "${WARP10_CONFIG_DIR}/00-warp.conf"
+  mv "${WARP10_CONFIG_DIR}/10-fdb.conf" "${WARP10_CONFIG_DIR}/10-fdb.conf.DISABLE"
+  getConfigFiles
+
+  ##
+  ## Retrieve LevelDB Home
+  ##
+  # shellcheck disable=SC2086
+  CONFIG_KEYS=$(${JAVACMD} -cp "${WARP10_JAR}" -Dfile.encoding=UTF-8 io.warp10.WarpConfig ${CONFIG_FILES} . 'leveldb.home' 2>/dev/null | grep -e '^@CONF@ ' | sed -e 's/^@CONF@ //')
```

Comment:
If we are initializing the config we do not need to read the existing files.

Reply:
removed